### PR TITLE
Adjust the cassandra example to use the service account.

### PR DESCRIPTION
**Pull request by brendandburns**
@lavalamp

### CODE REVIEW

**Comment by brendandburns** [comment]
Part of https://github.com/GoogleCloudPlatform/kubernetes/issues/7444

**Comment by brendandburns** [comment]
@erictune too.

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java [hunk 1]
```diff
@@ -65,21 +83,40 @@ public KubernetesSeedProvider(Map<String, String> params) {
...
-        String hostPort = getEnvOrDefault("KUBERNETES_RO_SERVICE_PORT", "8080");
-
-        String host = protocol + "://" + hostName + ":" + hostPort;
+        String host = "https://kubernetes.default.cluster.local";
         String serviceName = getEnvOrDefault("CASSANDRA_SERVICE", "cassandra");
         String path = "/api/v1beta3/namespaces/default/endpoints/";
         try {
+            String token = getServiceAccountToken();
+
+	    SSLContext ctx = SSLContext.getInstance("SSL");
```
Comment (lavalamp) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: indentation is inconsistent...

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java [hunk 2]
```diff
@@ -65,21 +83,40 @@ public KubernetesSeedProvider(Map<String, String> params) {
...
+
             URL url = new URL(host + path + serviceName);
+            HttpsURLConnection conn = (HttpsURLConnection)url.openConnection();
+	    conn.setSSLSocketFactory(ctx.getSocketFactory());
+	    conn.setHostnameVerifier(trustAllHosts);
+
+            conn.addRequestProperty("Authorization", "Bearer " + token);
             ObjectMapper mapper = new ObjectMapper();
-            Endpoints endpoints = mapper.readValue(url, Endpoints.class);
+            Endpoints endpoints = mapper.readValue(conn.getInputStream(), Endpoints.class);
```
Comment (lavalamp) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: Better way to do this is make a headless service & read the SRV records from DNS; no k8s api calls needed! Not sure if everything has landed for that to work yet though.

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java [hunk 3]
```diff
@@ -65,21 +83,40 @@ public KubernetesSeedProvider(Map<String, String> params) {
...
         try {
+            String token = getServiceAccountToken();
+
+	    SSLContext ctx = SSLContext.getInstance("SSL");
+	    ctx.init(null, trustAll, new SecureRandom());
+
             URL url = new URL(host + path + serviceName);
+            HttpsURLConnection conn = (HttpsURLConnection)url.openConnection();
+	    conn.setSSLSocketFactory(ctx.getSocketFactory());
+	    conn.setHostnameVerifier(trustAllHosts);
```
Comment (liggitt) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: TODO to remove this once we have a CA available?

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java (hunk 2 above)
Comment (brendandburns) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: Yeah, sadly there is no standard SRV record library for Java that I can find...  (spotify does have one that has been open sourced, though...)

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java (hunk 1 above)
Comment (brendandburns) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: fixed.  The perils of tab characters...

**Comment by brendandburns** [comment]
Comments addressed.  ptal.

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java (hunk 3 above)
Comment (brendandburns) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: done.

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java [hunk 4]
```diff
@@ -65,21 +83,43 @@ public KubernetesSeedProvider(Map<String, String> params) {
...
     public List<InetAddress> getSeeds() {
         List<InetAddress> list = new ArrayList<InetAddress>();
-        String protocol = getEnvOrDefault("KUBERNETES_API_PROTOCOL", "http");
-        String hostName = getEnvOrDefault("KUBERNETES_RO_SERVICE_HOST", "localhost");
-        String hostPort = getEnvOrDefault("KUBERNETES_RO_SERVICE_PORT", "8080");
-
-        String host = protocol + "://" + hostName + ":" + hostPort;
+        String host = "https://kubernetes.default.cluster.local";
         String serviceName = getEnvOrDefault("CASSANDRA_SERVICE", "cassandra");
         String path = "/api/v1beta3/namespaces/default/endpoints/";
```
Comment (lavalamp) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: I'm gonna merge this, but maybe in a followup you can fix this to get the namespace from the downward API? This forces the example to work only in the default namespace unnecessarily.

**Comment by lavalamp** [comment]
LGTM

File: examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java (hunk 2 above)
Comment (jimmidyson) [inline examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java]: @brendanburns We use dnsjava from http://www.xbill.org/dnsjava/ - very robust library indeed.

**Pull request merged**